### PR TITLE
WT-7503 Change the history store compressor from snappy to ZStandard.

### PR DESCRIPTION
**Pull request by bvpvamsikrishna**

### CODE REVIEW

**Comment by bvpvamsikrishna** [comment]
The failures are not related to the PR. The error's are on MacOS. They are:
2021/05/10 02:36:10.008] sysctl -n hw.logicalcpu
[2021/05/10 02:36:10.008] pwd
[2021/05/10 02:36:10.008] pwd
[2021/05/10 02:36:10.008] pwd
[2021/05/10 02:36:10.009] Command failed: command encountered problem: error waiting on process 'd52eda2b-c59b-4a22-ad5d-4197e4181ed0': exit status 1

File: src/include/btmem.h [hunk 1]
```diff
@@ -232,8 +232,8 @@ struct __wt_ovfl_reuse {
  * We also configure a larger than default internal page size to accommodate for larger history
  * store keys. We do that to reduce the chances of having to create overflow keys on the page.
  */
-#ifdef HAVE_BUILTIN_EXTENSION_SNAPPY
-#define WT_HS_COMPRESSOR "snappy"
+#ifdef HAVE_BUILTIN_EXTENSION_ZSTD
+#define WT_HS_COMPRESSOR "zstd"
```
Comment (kommiharibabu) [inline src/include/btmem.h:236]: Just to point that only the newer databases will use the `zstd`. All the existing versions still point to `snappy`. Is this what expected or the existing versions also want to switch?

Comment (bvpvamsikrishna) [reply src/include/btmem.h:236]: Yes, that is what is expected. we will not change the existing one's, only the new one's.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by bvpvamsikrishna** [comment]
@quchenhao, requesting your review.

**Review by quchenhao — APPROVED** [review]
lgtm

**Pull request merged**